cli/command.go

```diff
@@ -274,19 +278,42 @@ func (c *BaseCommand) Hidden() bool {
 
 // Prompt prompts the user for a value. If stdin is a tty, it prompts. Otherwise
 // it reads from the reader.
-func (c *BaseCommand) Prompt(msg string) (string, error) {
-	scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+func (c *BaseCommand) Prompt(ctx context.Context, msg string, defaultValue *string) (string, error) {
+	// guarantee Prompt is cancelable with SIGTERM and SIGINT
+	ctx, done := signal.NotifyContext(ctx, syscall.SIGTERM, syscall.SIGINT)
+	defer done()
 
-	if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
-		fmt.Fprint(c.Stdout(), msg)
-	}
+	resultChan := make(chan string)
+	errorChan := make(chan error)
+
+	go func() {
+		scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+
+		if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
+			fmt.Fprint(c.Stdout(), msg)
+		}
+
+		scanner.Scan()
+
+		if err := scanner.Err(); err != nil {
+			errorChan <- fmt.Errorf("failed to read stdin: %w", err)
+		}
+
+		if defaultValue != nil && strings.TrimSpace(scanner.Text()) == "" {
+			resultChan <- *defaultValue
+		}
 
-	scanner.Scan()
+		resultChan <- scanner.Text()
+	}()
 
-	if err := scanner.Err(); err != nil {
-		return "", fmt.Errorf("failed to read stdin: %w", err)
+	select {
+	case <-ctx.Done():
+		return "", fmt.Errorf("failed to prompt: %w", ctx.Err())
+	case err := <-errorChan:
+		return "", fmt.Errorf("failed to prompt: %w", err)
+	case result := <-resultChan:
+		return result, nil
```

Comment:
Like we discussed, this doesn't do what you think it does.

Reply:
Ill leave a comment.